test/trento_web/controllers/v1/host_controller_test.exs

```diff
@@ -48,4 +48,54 @@ defmodule TrentoWeb.V1.HostControllerTest do
              } == resp
     end
   end
+
+  describe "delete" do
+    test "should send 204 response when successful host deletion", %{conn: conn} do
+      %{id: host_id} = insert(:host)
+
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{host_id: ^host_id} ->
+          :ok
+        end
+      )
+
+      conn
+      |> delete("/api/v1/hosts/#{host_id}")
+      |> response(204)
+    end
+
+    test "should send 422 response if the host is still alive", %{conn: conn} do
+      %{id: host_id} = insert(:host)
+
+      expect(
+        Trento.Commanded.Mock,
+        :dispatch,
+        fn %Trento.Domain.Commands.RequestHostDeregistration{host_id: ^host_id} ->
+          {:error, :host_alive}
+        end
+      )
+
+      conn
+      |> delete("/api/v1/hosts/#{host_id}")
+      |> response(422)
```

Comment:
Same as the controller comment, we should use the correct schema for 422 as described in open api and we should check here with openapi spex test helper that the schema is respected